presto-main/src/main/java/com/facebook/presto/execution/QueryInfo.java

```diff
@@ -126,7 +129,8 @@ public QueryInfo(
             @JsonProperty("runtimeOptimizedStages") Optional<List<StageId>> runtimeOptimizedStages,
             @JsonProperty("addedSessionFunctions") Map<SqlFunctionId, SqlInvokedFunction> addedSessionFunctions,
             @JsonProperty("removedSessionFunctions") Set<SqlFunctionId> removedSessionFunctions,
-            @JsonProperty("planStatsAndCosts") StatsAndCosts planStatsAndCosts)
+            @JsonProperty("planStatsAndCosts") StatsAndCosts planStatsAndCosts,
+            @JsonProperty("planCanonicalInfo") List<CanonicalPlanWithInfo> planCanonicalInfo)
```

Comment:
The serde cost would be very expensive....

Reply:
I thought we only serialized after coverting to BasicQueryInfo, but maybe i was wrong. I removed this field from serialization, and added a test for that. Within coordinator, it should be ok to keep it

Reply:
remove the json property annotation then?